gc_registry/measurement/routes.py

```diff
@@ -1,17 +1,61 @@
 # Imports
 from esdbclient import EventStoreDBClient
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, HTTPException
 from sqlmodel import Session
 
 from gc_registry.core.database import db, events
 from gc_registry.measurement import models
+from gc_registry.measurement.services import parse_measurement_json
 
 # Router initialisation
 router = APIRouter(tags=["Measurements"])
 
 ### Device Meter Readings ###
 
 
+@router.post("/submit_readings", response_model=models.MeasurementSubmissionResponse)
+def submit_readings(
+    measurement_json: str,
+    write_session: Session = Depends(db.get_write_session),
+    read_session: Session = Depends(db.get_read_session),
+    esdb_client: EventStoreDBClient = Depends(events.get_esdb_client),
+):
+    """Submit meter readings as a JSON-serialised CSV file for one or more devices,
+    creating a MeasurementReport for each production interval against which GC
+    Bundles can be issued. Returns a summary of the readings submitted.
+
+    Args:
+        measurement_json (str): A JSON-serialised CSV file containing the meter readings.
+
+    Returns:
+        models.MeasurementSubmissionResponse: A summary of the readings submitted.
+    """
+    measurement_df = parse_measurement_json(measurement_json, to_df=True)
+
+    readings = models.MeasurementReport.create(
```

Comment:
Is this creation step necessary. If the registry stores duplicates of certificates in measurement data isn't that a bit redundant?

Can the incoming data just be read, validated and written to GCBs or respond with an Exception

Reply:
I mean the registry needs to keep the measurement data, even if it doesn't result in a GC issuance right?

Reply:
Will ask M-RETS but preference currently to not retain measurement records, any improperly issued GC rectification would need the device to send their records in again anyway. If keeping, consider retention period.